src/components/modules/BucketViewer/index.js

```diff
@@ -61,22 +62,26 @@ const BucketViewer = (props) => {
        * @param {AWS.S3.ListObjectsV2Output} response
        */
       const filterList = async (response, path) => {
-        const filetest = new RegExp(`^${path}([\\w!\\-\\.\\*'\\(\\), ]+[/]?)$`);
+        const filetest = new RegExp(
+          `^${path}([\\w!\\-\\.\\*'\\(\\), ]+[/]?)([\\w!\\-\\.\\*'\\(\\), ]+/)?`
+        );
         let newFiles = [];
-        const newFolders = [];
+        let newFolders = new Set([]);
         response.Contents.forEach((file) => {
           const filename = filetest.exec(file.Key);
           if (filename && filename[1]) {
             file.filename = filename[1];
-            if (filename[1][filename[1].length - 1] === '/') {
-              newFolders.push(file);
+            if (filename[2]) {
+              newFolders.add(filename[1]);
+            } else if (filename[1][filename[1].length - 1] === '/') {
+              newFolders.add(filename[1]);
             } else {
               newFiles.push(file);
             }
           }
         });
+        newFolders = [...newFolders].sort();
```

Comment:
Just to be sure if `newFolders` was a const it would not work since you are reassigning the variable?

Reply:
Yes exactly, it also is just a string array now.